odl/discr/tensor_ops.py

```diff
@@ -0,0 +1,610 @@
+﻿# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Operators defined for tensor fields."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+import numpy as np
+
+from odl.operator.operator import Operator
+from odl.set.sets import RealNumbers, ComplexNumbers
+from odl.space.fspace import FunctionSpace
+from odl.space.pspace import ProductSpace
+
+
+__all__ = ('PointwiseOperator', 'PointwiseNorm', 'PointwiseInner')
+
+
+class PointwiseOperator(Operator):
```

Comment:
Perhaps this name is somewhat too general. Some could interpret it as any "diagonal" operator X - > X. We may need to add something to the name to enforce that we mean for productspaces.